src/token.ts

```diff
@@ -119,9 +125,52 @@ export function buildParts(params: {
     payload.nnc = util.generateNonce()
   }
 
+  // Issuer key type must match UCAN algorithm
+  if (did.didToPublicKey(payload.iss).type !== keyType) {
+    throw new Error("The issuer's key type must match the given key type.")
+  }
+
+  // 📦
   return { header, payload }
 }
 
+/**
+ * Add a signature to a UCAN based on the given key pair.
+ */
+export async function sign(
+  header: UcanHeader,
+  payload: UcanPayload,
+  keypair: Keypair
```

Comment:
I've been working on `hs-ucan` recently, and noticed that we don't really need to pass in `header` here:
* We know the `typ` is always `JWT`
* We should set the version to `0.7.0` (or whatever the library generates) anyway, and we actually shouldn't allow users to customize this - since this might break the UCAN
* We know the JWT type from the `Keypair` *or* from the `DID` inside the payload

What do you think of removing the `header` here @icidasset?

But that's just an idle thought - if we do this it should be another PR anyway!

Reply:
We could remove it yeah.

I guess we then shouldn't allow to set a custom `ucanVersion` in `build` and `buildParts` neither?
And do we then still need `header` in the result from `buildParts`?
It might also make sense to rename the `sign` function to something else, since it's doing more than just signing the payload.

Reply:
> I guess we then shouldn't allow to set a custom ucanVersion in build and buildParts neither?

Yes. I think so.

>  And do we then still need header in the result from buildParts?

No, we wouldn't need that. And probably rename it to something like `buildPayload` or `buildClaims` (but if we go with `claims` we'd have to rename... many more things :P ).

>  It might also make sense to rename the sign function to something else, since it's doing more than just signing the payload.

Hmm. Maybe. If you find a good name? :smile:  I honestly can't come up with a good one right now :thinking: